xarray/convert.py

```diff
@@ -49,3 +62,111 @@ def set_cdms2_attrs(var, attrs):
     cdms2_var = cdms2.createVariable(var.values, axes=axes, id=dataarray.name)
     set_cdms2_attrs(cdms2_var, var.attrs)
     return cdms2_var
+
+
+# TODO: Add converting bounds from xarray to Iris and back
+def to_iris(dataarray):
+    """Convert a DataArray into a Iris Cube
+    """
+    # Iris not a hard dependency
+    import iris
+    try:
+        from iris.fileformats.netcdf import parse_cell_methods
+    except ImportError:
+        # prior to v1.10
+        from iris.fileformats._pyke_rules.compiled_krb.fc_rules_cf_fc \
+            import _parse_cell_methods as parse_cell_methods
+
+    # iris.unit is deprecated in Iris v1.9
+    import cf_units
+
+    def check_attrs(attrs, keys):
+        return dict((k, v) for k, v in attrs.items() if k in keys)
+
+    def get_args(attrs):
+        _args = {'attributes': filter_attrs(attrs, iris_forbidden_keys)}
```

Comment:
Also option is to make some private helper functions (preface the names
with _) in this module instead of nesting them inside the convert function.
That can often be significantly cleaner.

On Wed, May 11, 2016 at 1:30 PM, Neil Parley <notifications@github.com>
wrote:

> In xarray/convert.py
> <https://github.com/pydata/xarray/pull/814#discussion_r62921013>:
>
> > +    import iris
> > +    try:
> > +        from iris.fileformats.netcdf import parse_cell_methods
> > +    except ImportError:
> > +        # prior to v1.10
> > +        from iris.fileformats._pyke_rules.compiled_krb.fc_rules_cf_fc \
> > +            import _parse_cell_methods as parse_cell_methods
> > +
> > +    # iris.unit is deprecated in Iris v1.9
> > +    import cf_units
> > +
> > +    def check_attrs(attrs, keys):
> > +        return dict((k, v) for k, v in attrs.items() if k in keys)
> > +
> > +    def get_args(attrs):
> > +        _args = {'attributes': filter_attrs(attrs, iris_forbidden_keys)}
>
> It was because there was an outer args and I wanted to make it clear this
> was not that. I have seen people write helper functions and make use of
> outer scope variables. Again maybe it's clear enough here.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/xarray/pull/814/files/cd06a2e23e77fff6c96ad00c6285a5966bc63b1f#r62921013>
>